prov/verbs/src/fi_verbs.h

```diff
@@ -148,6 +155,80 @@ struct verbs_addr {
 	struct rdma_addrinfo *rai;
 };
 
+/*
+ * fields of Infiniband packet headers that are used to
+ * represent OFI EP address
+ * - LRH (Local Route Header) - Link Layer:
+ *   - LID - destination Local Identifier
+ *   - SL - Service Level
+ * - GRH (Global Route Header) - Network Layer:
+ *   - GID - destination Global Identifier
+ * - BTH (Base Transport Header) - Transport Layer:
+ *   - QPN - destination Queue Oair number
+ *   - P_key - Partition Key
+ *
+ * Note: DON'T change the placement of the fields in the structure.
+ *       The placement is to keep structure size = 256 bits (32 byte).
+ */
+struct ofi_ib_ud_ep_name {
+	union ibv_gid	gid;		/* 64-bit GUID + 64-bit EUI - GRH */
+
+	uint32_t	qpn;		/* BTH */
+
+	uint16_t	lid; 		/* LRH */
+	uint16_t	pkey;		/* BTH */
+	uint16_t	service;	/* for NS src addr, 0 means any */
+
+	uint8_t 	sl;		/* LRH */
+	uint8_t		padding[3];	/* forced padding to 256 bits (32 byte) */
+
+	uint16_t	checksum;	/* to check compliance of ep_name */
```

Comment:
Adding a checksum to the name is weird.  Checksum should apply to packets, not a data structure.

Reply:
Yes, I completely agree with you.
we needed a mechanism that will be used to identify the address is whether `struct ofi_ib_ud_ep_name`
But your next comment helps me to understand that the mentioned mechanism isn't needed anymore